code/obj/machinery/computer/robotics.dm

```diff
@@ -22,6 +37,121 @@
 		..()
 		STOP_TRACKING
 
+	ui_interact(mob/user, datum/tgui/ui)
+		ui = tgui_process.try_update_ui(user, src, ui)
+		if (!ui)
+			ui = new(user, src, "RoboticsControl")
+			ui.open()
+
+	ui_data(mob/user)
+		src.update_silicon_statuses()
+		src.update_ghostdrone_statuses()
+		. = list(
+			"user_is_ai" = isAI(user),
+			"user_is_cyborg" = isrobot(user),
+			"ai_names" = src.ai_names,
+			"ai_statuses" = src.ai_statuses,
+			"ai_killswitch_times" = src.ai_killswitch_times,
+			"cyborg_names" = src.cyborg_names,
+			"cyborg_statuses" = src.cyborg_statuses,
+			"cyborg_cell_charges" = src.cyborg_cell_charges,
+			"cyborg_modules" = src.cyborg_modules,
+			"cyborg_lock_times" = src.cyborg_lock_times,
+			"cyborg_killswitch_times" = src.cyborg_killswitch_times,
+			"ghostdrone_names" = src.ghostdrone_names
+		)
+
+	ui_act(action, params)
+		. = ..()
+		if (.)
+			return
+		. = TRUE
```

Comment:
Do on a case-by-case basis, please. It might seem redundant, but applying it here regardless is not generally how we do it for the sake of making updating the UI a conscious choice. Just looking at your flow, a weapons lock on an emagged cyborg doesn't have any reason to update the UI, for example.